### ISSUE TITLE
Enable envoy track_remaining metrics

### ISSUE THREAD

**Issue by rafaelcoelho** [istio/istio#41779]
Hi Guys,

I'm struggling to configure the Envoy proxy from Istio (**1.9.4 version**) in order to enable the `track_remaining` [flag](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/circuit_breaker.proto#envoy-v3-api-field-config-cluster-v3-circuitbreakers-thresholds-track-remaining).

The circuit breaker threshold settings have being done via `DestinationRule`:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: my-custom-dr
  namespace: default
spec:
  host: >-
    my-custom-workload-service.my-custom-workload-api.svc.cluster.local
  subsets:
    - labels:
        app: my-custom-workload
        app.kubernetes.io/instance: my-custom-workload
      name: canary
    - labels:
        app: my-custom-workload
        app.kubernetes.io/instance: my-custom-workload
      name: stable
  trafficPolicy:
    connectionPool:
      http:
        maxRequestsPerConnection: 10
      tcp:
        maxConnections: 1
```

And this is the `EnvoyFilter` used to enable the `track_remaining` flag:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: my-custom-workload-track-remaining
  namespace: default
spec:
  configPatches:
    - applyTo: CLUSTER
      match:
        cluster:
          service: >-
            my-custom-workload-service.my-custom-workload-api.svc.cluster.local
      patch:
        operation: MERGE
        value:
          circuit_breakers:
            thresholds:
              - track_remaining: true
```

However, from rollout Envoy dump configuration the following is placed:
```json
"cluster": {
  "@type": "type.googleapis.com/envoy.config.cluster.v3.Cluster",
  "circuit_breakers": {
    "thresholds": [
      {
        "max_connections": 1,
        "max_pending_requests": 4294967295,
        "max_requests": 4294967295,
        "max_retries": 4294967295
      },
      {
        "track_remaining": true
      }
    ]
  },
  "connect_timeout": "10s"
```

Where the desired result would be:
```json
"cluster": {
  "@type": "type.googleapis.com/envoy.config.cluster.v3.Cluster",
  "circuit_breakers": {
    "thresholds": [
      {
        "max_connections": 1,
        "max_pending_requests": 4294967295,
        "max_requests": 4294967295,
        "max_retries": 4294967295,
        "track_remaining": true
      }
    ]
  },
  "connect_timeout": "10s"
```

Am I missing something or the flag set shall be done in another manifest apart from `EnvoyFilter`?

**Comment by rafaelcoelho** [comment]
Looks like the default `true` value have been added from PR #31807 released from 1.10.0 [a49a2fa61](https://github.com/istio/istio/commit/a49a2fa614024db27b0a02a68c8ec97eb2e1bf2f) that is not into 1.9.4 version.

@ramaraochavali, do you have any insight how to set the flag for `true' upon the versions that doesn't have the default value (like the 1.9.4)?

Thanks in advanced

**Comment by ramaraochavali** [comment]
I guess it is because thresholds is an array - it is being merged as another element.  Do not know from top of my head how to merge an element in an array.

On a side note, 1.9 is deprecated.
